src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/client/FuncTestRequestHandler.java

```diff
@@ -183,12 +186,18 @@ private RequestSpecification withAuthenticatedRequest() {
     }
 
     private String getBearerToken() {
-        IdamOpenIdClient idamOpenIdClient = new IdamOpenIdClient(testConfig);
+        if (null == idamOpenIdClient) {
+            idamOpenIdClient = new IdamOpenIdClient(testConfig);
+            log.info("IdamOpenIdClient client:::" + idamOpenIdClient.getBearerToken());
+        }
         return idamOpenIdClient.getBearerToken();
```

Comment:
idamOpenIdClient.getBearerToken() is expensive call and this is called each time when the method is called.Needs optimisation.